test/metabase/api/action_test.clj

```diff
@@ -633,3 +639,99 @@
           (testing "error if actions is disabled"
             (is (= "Actions are not enabled."
                  (:message (mt/user-http-request :crowberto :get 400 (format "action/%d/execute" delete-action-id) :parameters (json/encode {:id 1})))))))))))
+
+(mt/defdataset action-error-handling
+  [["group"
+    [{:field-name "name" :base-type :type/Text :not-null? true}
+     {:field-name "ranking" :base-type :type/Integer :not-null? true :unique? true}]
+    [["admin" 1]
+     ["user" 2]]]
+   ["user"
+    [{:field-name "name" :base-type :type/Text :not-null? true}
+     {:field-name "group-id" :base-type :type/Integer :fk "group" :not-null? true}]
+    [["crowberto" 1]
+     ["rasta"     2]
+     ["lucky"     1]]]])
+
+(deftest action-error-handling-test
+  (mt/test-drivers (mt/normal-drivers-with-feature :actions)
+    (mt/dataset action-error-handling
+      (mt/with-actions-enabled
+        (mt/with-current-user (mt/user->id :crowberto)
+          (mt/with-actions [{_card-id :id}             {:dataset_query (mt/mbql-query group) :dataset true}
+                            {create-group :action-id} {:type :implicit
+                                                       :kind "row/create"}
+                            {update-group :action-id} {:type :implicit
+                                                       :kind "row/update"}
+                            {delete-group :action-id} {:type :implicit
+                                                       :kind "row/delete"}]
+            (mt/with-actions [{_card-id :id}           {:dataset_query (mt/mbql-query user) :dataset true}
+                              {create-user :action-id} {:type :implicit
+                                                        :kind "row/create"}
+                              {update-user :action-id} {:type :implicit
+                                                        :kind "row/update"}]
+              (testing "violate not-null constraint"
+                (testing "when creating"
+                  (is (= {:message "Ranking must have values."
+                          :errors {:ranking "You must provide a value."}}
```

Comment:
yea, `column-errors` might be better described that, but I have 2 counter points:
- errors are also used in the API to return `per param schema error`, so that's might be an existing pattern?
- our middleware have a special treatment for `:errors` though ([here](https://github.com/metabase/metabase/blob/ngoc-actions-error-handling-cleanup/src/metabase/server/middleware/exceptions.clj#L65)), so if we change to `column-errors` we'd need to treat it a bit different. Also this is an related [issue](https://github.com/metabase/metabase/issues/14683)

Reply:
Fair points, lets keep the `:errors` key. I wasn't aware that's how the middleware works.